source/gateway.go

```diff
@@ -300,9 +295,16 @@ func (sc *gatewaySource) hostNamesFromGateway(gateway networkingv1alpha3.Gateway
 				host = parts[1]
 			}
 
-			hostnames = append(hostnames, host)
+			if host != "*" {
```

Comment:
can you club this check with `if host == "" {` in L286?

Reply:
I don't think so - `my-namespace/*` is a valid option for the host, so we need to check for `*` after stripping the namespace.

Reply:
ah yes makes sense :)